addons/account/data/bank_statement_balance.sql

```diff
@@ -0,0 +1,20 @@
+CREATE OR REPLACE FUNCTION anchor_accumulator
+-- Calculate the sum of a list of numbers, replacing the calculated value with the anchor(check) points when available.
+    (
+    cur_bal NUMERIC,
+    amount NUMERIC,
+    is_anchor BOOL,
+    anchor_value NUMERIC
+    )
+    RETURNS NUMERIC AS
+$$
+SELECT (CASE WHEN is_anchor = TRUE THEN anchor_value ELSE cur_bal+amount END);
+$$ LANGUAGE 'sql' CALLED ON NULL INPUT;
+
+CREATE OR REPLACE AGGREGATE statement_line_balance(NUMERIC,BOOL,NUMERIC)
+(
+    INITCOND = 0,
+    STYPE = NUMERIC,
+    SFUNC = anchor_accumulator
+);
+
```

Comment:
There is a test, I mentioned you on it.
It is not a complicated function, it is just conditional sum, *if this is an anchor point, use it as the sum value (because it has been validated by the user, and they insist on having this value as sum) otherwise, sum it up normally*

Reply:
I did not get what you mean by initial balance and real statement. 
internal_index is for sorting the lines without hassle 
for is_anchor, directly working with null value is a bit risky, if a statement reset to draft (un-validated)  we have to nullify this field and how can we do it with the orm? 
I think having a boolean field is more readable and visible.
The sql function is to be removed because nobody has complained about the speed in here, so we do it in python and we switch to this function if anyone complains about the speed.